deviations/templates/deviations/list_dl.html

```diff
@@ -22,40 +22,90 @@
 	<div class="panel-heading">
 		<h3 class="panel-title">{% translate "DEADLINE_DEVIATIONS" %}</h3>
 	</div>
-	<table class="table table-striped table-bordered table-condensed">
-		<tr>
-			<th>{% translate "SUBMITTER" %}</th>
-			<th>{% translate "EXERCISE" %}</th>
-			<th>{% translate "EXTRA_MINUTES" %}</th>
-			<th>{% translate "WITHOUT_LATE_PENALTY" %}</th>
-			<th>{% translate "GRANTER" %}</th>
-			<th>{% translate "GRANT_TIME" %}</th>
-			<th>{% translate "ACTIONS" %}</th>
-		</tr>
-		{% translate "UNKNOWN" as unknown %}
-		{% for deviation in deviations %}
-		<tr>
-			<td>{{ deviation.submitter }}</td>
-			<td>{{ deviation.exercise }}</td>
-			<td>{{ deviation.extra_minutes }}</td>
-			<td>{{ deviation.without_late_penalty }}</td>
-			<td>{{ deviation.granter|default_if_none:unknown }}</td>
-			<td>{{ deviation.grant_time|default_if_none:unknown }}</td>
-			<td>
-				<form method="post" action="{{ deviation|url:'deviations-remove-dl' }}">
-					{% csrf_token %}
-					<button type="submit" class="aplus-button--danger aplus-button--xs">
-						<span class="glyphicon glyphicon-remove" aria-hidden="true"></span>
-						{% translate "REMOVE" %}
-					</button>
-				</form>
-			</td>
-		</tr>
-		{% empty %}
-		<tr>
-			<td colspan="4">{% translate "NO_DEADLINE_DEVIATIONS" %}</td>
-		</tr>
-		{% endfor %}
+	<table class="table table-striped table-bordered table-condensed filtered-table ordered-table grouped-table">
+		<thead>
+			<tr>
+				<th>{% translate "SUBMITTER" %}</th>
+				<th>{% translate "EXERCISE" %}</th>
+				<th>{% translate "EXTRA_MINUTES" %}</th>
+				<th
+					data-filter-type="options"
+					data-filter-options="{% translate 'YES' %}|{% translate 'NO' %}"
+					data-order-disable="true"
+				>
+					{% translate "WITHOUT_LATE_PENALTY" %}
+				</th>
+				<th>{% translate "GRANTER" %}</th>
+				<th>{% translate "GRANT_TIME" %}</th>
+				<th data-filter-type="none" data-order-disable="true">{% translate "ACTIONS" %}</th>
+			</tr>
+		</thead>
+		<tbody>
+			{% translate "UNKNOWN" as unknown %}
+			{% for deviations, can_group, group_id in deviation_groups %}
+			{% if can_group %}
+			<tr data-group-parent="{{ group_id }}">
+				<td>{{ deviations.0.submitter }}</td>
+				<td>
+					{{ deviations.0.exercise.course_module }}
+					<strong>
+						({% blocktranslate trimmed with count=deviations|length %}
+						NUM_OF_EXERCISES -- {{ count }}
+						{% endblocktranslate %})
```

Comment:
Django translations support pluralization based on `count`: 1 exercise, 2 exercises

https://docs.djangoproject.com/en/3.2/topics/i18n/translation/#pluralization

Reply:
True, but a group must always have more than one deviation (and therefore more than one exercise).